api/wasm/cpp/proxy_wasm_externs.h

```diff
@@ -85,6 +86,18 @@ extern "C" void proxy_getSharedData(const char* key_ptr, size_t key_size, const
 extern "C" bool proxy_setSharedData(const char* key_ptr, size_t key_size, const char* value_ptr,
                                     size_t value_size, uint32_t cas);
 
+
+// SharedQueue
+// Note: queue_name must be globally unique.  Registering the same queue_name will overwrite the old registration while
+// preseving any pending data.  Consequently it should typically be followed by a call to proxy_dequeueSharedQueue.
```

Comment:
Why overwrite _and_ preserve old data?

Also, how are extensions expected to guarantee uniqueness when VM can be shared by many of them?

Reply:
We need to be able to send messages between VMs so there must be a global namespace, however, we could require that the queue_name be (vm_id, queue_name).  That would prevent collisions and would prevent stealing of someone elses queue.

Reply:
Sounds reasonable.

Reply:
We want to preserve the existing data in case there is a reconfigure which causes a VM to restart and we don't want to lose any data.